pkg/cmd/repo/create/create.go

```diff
@@ -349,90 +333,324 @@ func createRun(opts *CreateOptions) error {
 		templateRepoMainBranch = repo.DefaultBranchRef.Name
 	}
 
-	createLocalDirectory := opts.ConfirmSubmit
-	if !opts.ConfirmSubmit {
-		opts.ConfirmSubmit, err = confirmSubmission(input.Name, input.OwnerLogin, inLocalRepo)
+	repo, err := repoCreate(httpClient, repoToCreate.RepoHost(), input)
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		doCreate, err := confirmSubmission(opts.Name, repoToCreate.RepoOwner(), opts.Visibility)
 		if err != nil {
 			return err
 		}
+		if !doCreate {
+			return nil
```

Comment:
If the user has chosen not to submit, the command should exit with non-zero code.